src/Components/Overlay/index.jsx

```diff
@@ -1,11 +1,50 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./Overlay.scss";
+import { DICTIONARY_URL } from "../../constants";
+import axios from "axios";
+import { fetchData } from "../../util";
+
+const Overlay = ({ word, setIsOverlayVisible }) => {
+  const [definitions, setDefinition] = useState([]);
+  console.log(definitions);
+
+  useEffect(() => {
+    const getWordDefinition = async () => {
+      const url = DICTIONARY_URL + word;
+      let { succeeded, response } = await fetchData(url);
+      if (succeeded) {
+        response.map((data) => {
+          return data.meanings.map((each) =>
+            each.definitions.map((desc) =>
+              setDefinition((value) => [...value, desc.definition])
+            )
+          );
+        });
```

Comment:
This will populate duplicates if rendered twice. Instead of spreading initial values, create a new empty array and map your definitions inside it every time.